nodedb.go

```diff
@@ -845,14 +850,13 @@ func (ndb *nodeDB) uncacheFastNode(key []byte) {
 
 // Add a node to the cache and pop the least recently used node if we've
 // reached the cache size limit.
-// CONTRACT: the caller must serizlize access to this method through ndb.mtx.
-func (ndb *nodeDB) cacheFastNode(node *FastNode) {
+func (ndb *nodeDB) cacheFastNode(node Node) {
 	elem := ndb.fastNodeCacheQueue.PushBack(node)
-	ndb.fastNodeCache[string(node.key)] = elem
+	ndb.fastNodeCache[string(node.Key())] = elem
 
 	if ndb.fastNodeCacheQueue.Len() > ndb.fastNodeCacheSize {
 		oldest := ndb.fastNodeCacheQueue.Front()
-		key := ndb.fastNodeCacheQueue.Remove(oldest).(*FastNode).key
+		key := ndb.fastNodeCacheQueue.Remove(oldest).(Node).Key()
```

Comment:
can we avoid type casts everywhere? Imo it is better to have a concrete type than having to cast